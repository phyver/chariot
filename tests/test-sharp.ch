-- :set allow_structs true

codata t where D : t -> t

val f a b c d = D# a b c d

val g a b c d = D# a $ b $ c d

val h a = D # D # D # a

:show functions
