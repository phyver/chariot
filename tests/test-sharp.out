>> option verbose is set to 1
>> option allow_structs is set to true
>> coinductive type t was succesfully defined
>> Typing for f successful
>> the definition for f is complete
>> the definition f is provably correct

>> Typing for g successful
>> the definition for g is complete
>> the definition g is provably correct

>> Typing for h successful
>> the definition for h is complete
>> the definition h is provably correct

>> functions in environment:
val
   f : ('a → 'b → 'c → t) → 'a → 'b → 'c → t
    | f a b c d = { D = a b c d }⁰

val
   g : ('a → t) → ('b → 'a) → ('c → 'b) → 'c → t
    | g a b c d = { D = a (b (c d)) }⁰

val
   h : t → t
    | h a = { D = { D = { D = a }⁰ }⁰ }⁰

